RubberduckTests/Refactoring/ExtractMethod/ExtractMethodModelTests.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Refactorings.ExtractMethod;
+using Rubberduck.VBEditor;
+
+namespace RubberduckTests.Refactoring.ExtractMethod
+{
+    [TestClass]
+    public class ExtractMethodModelTests
+    {
+
+        [TestClass]
+        public class when_declarations_contain_no_previous_newMethod
+        {
+            [TestMethod]
+            public void should_return_newMethod()
+            {
+
+                QualifiedModuleName qualifiedModuleName;
+                RubberduckParserState state;
+                var inputCode = @"
+Option Explicit
+Private Sub Foo()
+    Dim x As Integer
+    x = 1 + 2
+End Sub";
+
+                MockParser.ParseString(inputCode, out qualifiedModuleName, out state);
+                var declarations = state.AllDeclarations;
+                var selection = new Selection(4, 4, 4, 14);
+                QualifiedSelection? qSelection = new QualifiedSelection(qualifiedModuleName, selection);
+
+                var SUT = new ExtractMethodModel(declarations, qSelection.Value, "x = 1 + 2");
+
+                var actual = SUT.Method.MethodName;
+                var expected = "NewMethod";
+
+                Assert.AreEqual(expected, actual);
+
+            }
+
+        }
+
+        [TestClass]
+        public class when_declarations_contain_a_previous_newMethod
```

Comment:
Rubberduck tries to adhere to [standard C# naming conventions](https://msdn.microsoft.com/en-us/library/x2dbyw72(v=vs.71).aspx). As such, class and method names should be `PascalCase`, including test classes and test methods.

I love the `SUT`, `actual` and `expected` identifers though, and the tests themselves are nice and clear.